app/jobs/assignment_repo/porter_status_job.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+class AssignmentRepo
+  class PorterStatusJob < ApplicationJob
+    class ImportInProgress < StandardError; end
+    queue_as :porter_status
+    retry_on Octopoller::TimeoutError, queue: :porter_status
+
+    # rubocop:disable MethodLength
+    def perform(assignment_repo, user)
+      github_repository = assignment_repo.github_repository
+
+      result = Octopoller.poll do # TODO: log errors
```

Comment:
Should the `Octopoller` API accept a `retry_on_error:` argument that accepts an error to re-poll on? Right now `Octopoller` swallows any error and re-polls.